packages/fluentui/react-northstar/src/components/Dialog/Dialog.tsx

```diff
@@ -117,304 +132,334 @@ export const dialogSlotClassNames: DialogSlotClassNames = {
+  setStart();
+
+  const {
+    accessibility,
+    content,
+    header,
+    actions,
+    cancelButton,
+    closeOnOutsideClick,
+    confirmButton,
+    headerAction,
+    overlay,
+    trapFocus,
+    trigger,
+    footer,
+    backdrop,
+    className,
+    design,
+    styles,
+    variables,
+  } = props;
+
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(Dialog.handledProps, props);
+
+  const contentRef = React.useRef<HTMLElement>();
+  const overlayRef = React.useRef<HTMLElement>() ;
+  const triggerRef = React.useRef<HTMLElement>();
+
+  const contentId = React.useRef<string>();
+  contentId.current = getOrGenerateIdFromShorthand('dialog-content-', content, contentId.current);
+  const headerId = React.useRef<string>();
+  headerId.current = getOrGenerateIdFromShorthand('dialog-header-', header, headerId.current);
+
+  const getA11yProps = useAccessibility<DialogBehaviorProps>(accessibility, {
+    debugName: Dialog.displayName,
+    actionHandlers: {
+      closeAndFocusTrigger: e => {
+        handleDialogCancel(e);
+        e.stopPropagation();
+
+        _.invoke(triggerRef, 'current.focus');
+      },
+      close: e => handleDialogCancel(e),
+    },
+    mapPropsToBehavior: () => ({
+      headerId: headerId.current,
+      contentId: contentId.current,
+      trapFocus,
+      trigger,
+    }),
+    rtl: context.rtl,
+  });
 
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      children: false,
-      content: 'shorthand',
+  const { classes, styles: resolvedStyles } = useStyles<DialogStylesProps>(Dialog.displayName, {
+    className: dialogClassName,
+    mapPropsToStyles: () => ({
+      backdrop,
     }),
-    actions: customPropTypes.itemShorthand,
-    backdrop: PropTypes.bool,
-    headerAction: customPropTypes.itemShorthand,
-    cancelButton: customPropTypes.itemShorthand,
-    closeOnOutsideClick: PropTypes.bool,
-    confirmButton: customPropTypes.itemShorthand,
-    defaultOpen: PropTypes.bool,
-    header: customPropTypes.itemShorthand,
-    onCancel: PropTypes.func,
-    onConfirm: PropTypes.func,
-    onOpen: PropTypes.func,
-    open: PropTypes.bool,
-    overlay: customPropTypes.itemShorthand,
-    trapFocus: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
-    trigger: PropTypes.any,
-  };
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-  static defaultProps = {
-    accessibility: dialogBehavior as Accessibility,
-    actions: {},
-    backdrop: true,
-    closeOnOutsideClick: true,
-    overlay: {},
-    footer: {},
-    trapFocus: true,
-  };
+  const [open, setOpen] = useAutoControlled({
+    defaultValue: props.defaultOpen,
+    value: props.open,
+    initialValue: false,
+  });
 
-  static autoControlledProps = ['open'];
-  static Footer = DialogFooter;
+  const lockBodyScroll = React.useCallback(() => {
```

Comment:
Move out these functions out of render, they don't need to be wrapped with `useCallback` as they have a single param `context.target`

Reply:
Moved to utils